src/main/python/inverse_parser/_node_restorer.py

```diff
@@ -0,0 +1,684 @@
+        xml_node_children_comps = _XmlNodeChildrenGetter.get_children(xml_node, with_tag='comprehension')
+        py_node.generators = _NodeRestorer.restore_many(xml_node_children_comps)
+
+    @staticmethod
+    def restore_ast_comprehension(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.target = _NodeRestorer.restore(xml_node_children[0])
+        py_node.iter = _NodeRestorer.restore(xml_node_children[1])
+        py_node.ifs = _NodeRestorer.restore_many(xml_node_children[2:])
+        #  py_node.is_async = ? TODO: parser doesnt account async's
+
+    # - Statements:
+
+    @staticmethod
+    def restore_ast_assign(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node_children = _NodeRestorer.restore_many(xml_node_children)
+        py_node.targets = py_node_children[:-1]
+        py_node.value = py_node_children[-1]
+
+    @staticmethod
+    def restore_ast_ann_assign(xml_node: ET.Element, py_node: ast.AST) -> None:  # TODO: astmonkey.to_source doesn’t support AnnAssign
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.target = _NodeRestorer.restore(xml_node_children[0])
+        py_node.annotation = _NodeRestorer.restore(xml_node_children[1])
+        py_node.value = _NodeRestorer.restore(xml_node_children[2])
+        #  py_node.simple = ? TODO: parser doesnt account "simple" field
+
+    @staticmethod
+    def restore_ast_aug_assign(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.target = _NodeRestorer.restore(xml_node_children[0])
+        py_node.value = _NodeRestorer.restore(xml_node_children[1])
+
+    @staticmethod
+    def restore_ast_raise(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.exc = _NodeRestorer.restore(xml_node_children[0])
+        try:
+            py_node.cause = _NodeRestorer.restore(xml_node_children[1])
+        except IndexError:
+            py_node.cause = None
+
+    @staticmethod
+    def restore_ast_assert(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.test = _NodeRestorer.restore(xml_node_children[0])
+        try:
+            py_node.msg = _NodeRestorer.restore(xml_node_children[1])
+        except IndexError:
+            py_node.msg = None
+
+    @staticmethod
+    def restore_ast_delete(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.targets = _NodeRestorer.restore_many(xml_node_children)
+
+    @staticmethod
+    def restore_ast_pass(xml_node: ET.Element, py_node: ast.AST) -> None:
+        pass
```

Comment:
Is it ok? Should we implement it? If it is not, describe why in a comment, please

Reply:
Yes, it's ok. This is the implementation